pkg/kubelet/kubelet.go

```diff
@@ -2323,32 +2324,64 @@ func (kl *Kubelet) rejectPod(pod *api.Pod, reason, message string) {
 		Message: "Pod " + message})
 }
 
+// getNodeAnyWay() must return a *api.Node which is required by RunGeneralPredicates().
+// The *api.Node is obtained as follows:
+// 1. When kubelet is in standaloneMode, manufactures *api.Node with kubelet
+// 2. Try to get *api.Node from kubelet's nodeInfo, if failed, we try 3.
+// 3. Manufactures *api.Node with kubelet
+func (kl *Kubelet) getNodeAnyWay() (*api.Node, error) {
+	if kl.standaloneMode {
+		return kl.initialNodeStatus()
+	}
+	if n, err := kl.nodeInfo.GetNodeInfo(kl.nodeName); err == nil {
+		return n, nil
+	}
+	return kl.initialNodeStatus()
+}
+
 // canAdmitPod determines if a pod can be admitted, and gives a reason if it
 // cannot. "pod" is new pod, while "pods" include all admitted pods plus the
 // new pod. The function returns a boolean value indicating whether the pod
 // can be admitted, a brief single-word reason and a message explaining why
 // the pod cannot be admitted.
-//
-// This needs to be kept in sync with the scheduler's and daemonset's fit predicates,
-// otherwise there will inevitably be pod delete create loops. This will be fixed
-// once we can extract these predicates into a common library. (#12744)
 func (kl *Kubelet) canAdmitPod(pods []*api.Pod, pod *api.Pod) (bool, string, string) {
-	if hasHostPortConflicts(pods) {
-		return false, "HostPortConflict", "cannot start the pod due to host port conflict."
-	}
-	if !kl.matchesNodeSelector(pod) {
-		return false, "NodeSelectorMismatching", "cannot be started due to node selector mismatch"
-	}
-	cpu, memory := kl.hasInsufficientfFreeResources(pods)
-	if cpu {
-		return false, "InsufficientFreeCPU", "cannot start the pod due to insufficient free CPU."
-	} else if memory {
-		return false, "InsufficientFreeMemory", "cannot be started due to insufficient free memory"
+	node, err := kl.getNodeAnyWay()
+	if err != nil {
+		glog.Warningf("Cannot get Node info: %v", err)
```

Comment:
This should probably be `glog.Errorf` since not being able to admit pod is a serious issue.